src/ucp/dt/datatype_iter.c

```diff
@@ -72,6 +72,10 @@ void ucp_datatype_iter_iov_mem_dereg(ucp_context_h context,
     ucp_mem_h *memh = dt_iter->type.iov.memh;
     size_t iov_index, length;
 
+    if (memh == NULL) {
```

Comment:
how could the meme **array** be NULL?

Reply:
when it is not registered, plz check line 47 in this file

Reply:
ok